### PR TITLE
Remove Mixer + RBAC + 3Scale deps

### PR DESCRIPTION
**Pull request by lucasponce**
Kiali is going to stop using some old/deprecated APIs so it can be removed from the roles for future versions.
Also, removing 3scale config as this extensions depends on Mixer module.

### CODE REVIEW

**Review by xeviknal — COMMENTED** [review]
I think there are other files to be updated here. Mazz always points to the check-list when a change in this area is done: https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-role-permission

I think that the helm-charts and the openshift roles needs to be updated too.

**Review by hhovsepy — APPROVED** [review]
Verified

File: roles/default/kiali-deploy/templates/kubernetes/role-viewer.yaml [hunk 1]
```diff
@@ -47,10 +47,7 @@ rules:
   - list
   - watch
 - apiGroups:
-  - config.istio.io
   - networking.istio.io
-  - authentication.istio.io
-  - rbac.istio.io
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/kubernetes/role-viewer.yaml:53]: I think you also want to do this to the openshift role-viewer.yaml template.  This: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/openshift/role-viewer.yaml#L49-L54

File: roles/default/kiali-deploy/templates/kubernetes/role.yaml [hunk 2]
```diff
@@ -50,10 +50,7 @@ rules:
   - patch
   - watch
 - apiGroups:
-  - config.istio.io
   - networking.istio.io
-  - authentication.istio.io
-  - rbac.istio.io
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/kubernetes/role.yaml:56]: I think you also want to do this to the openshift role.yaml template. This: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/openshift/role.yaml#L52-L57

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Review by jmazzitelli — APPROVED** [review]
LGTM

**Pull request merged**